crates/rpc/src/v1/impls/parity.rs

```diff
@@ -268,16 +268,28 @@ where
             .map(Into::into)
     }
 
-    fn pending_transactions(&self, limit: Option<usize>) -> Result<Vec<Transaction>> {
+    fn pending_transactions(
+        &self,
+        limit: Option<usize>,
+        filter: Option<TransactionFilter>,
+    ) -> Result<Vec<Transaction>> {
         let ready_transactions = self.miner.ready_transactions(
             &*self.client,
-            limit.unwrap_or_else(usize::max_value),
+            usize::max_value(),
```

Comment:
you omitted `limit` here?

Reply:
This will always copy every pending tx from pool and then do the filtering below.